crates/sui-framework/packages/sui-framework/sources/coin_metadata_registry.move

```diff
@@ -0,0 +1,208 @@
+module sui::coin_metadata_registry;
+
+use std::string::String;
+use sui::balance::Supply;
+use sui::coin::{TreasuryCap, CoinMetadata, RegulatedCoinMetadata};
+use sui::dynamic_field;
+use sui::transfer::Receiving;
+
+const ENotSystemAddress: u64 = 0;
+const EMetadataAlreadyExists: u64 = 1;
+const EMetadataNotFound: u64 = 2;
+const EAlreadyClaimed: u64 = 3;
+const EMetadataCapNotClaimed: u64 = 4;
+const EAlreadyRegulated: u64 = 5;
+
+public struct Registry has key { id: UID }
```

Comment:
nit: the name should match the purpose, maybe something like `CoinMetadataRegistry` would do?